From 20baf9fb3af7c7b78f5c5e8af36b4bb886c695d3 Mon Sep 17 00:00:00 2001
From: "Joshua M. Clulow" <jmc@joyent.com>
Date: Sun, 23 Jul 2017 14:30:33 +0000
Subject: [PATCH] MANTA-3192 kick_off_audit crashes when previous job's output
 was too large Reviewed by: David Pacheco <dap@joyent.com> Approved by: David
 Pacheco <dap@joyent.com>

---
 bin/kick_off_audit.js | 50 ++++++++++++++++++++++++++++++++-----------
 1 file changed, 37 insertions(+), 13 deletions(-)

diff --git a/bin/kick_off_audit.js b/bin/kick_off_audit.js
index edd3003..7e6e39d 100755
--- a/bin/kick_off_audit.js
+++ b/bin/kick_off_audit.js
@@ -16,6 +16,9 @@ var getopt = require('posix-getopt');
 var lib = require('../lib');
 var manta = require('manta');
 var path = require('path');
+var verror = require('verror');
+
+var VE = verror.VError;
 
 
 
@@ -259,21 +262,42 @@ function checkJobResults(job, audit, opts, cb) {
                         return (cb(null));
                 }
 
-                gopts.path = parts[0];
-                lib.common.getObject(gopts, function (err2, errorLines) {
-                        if (err2) {
-                                //Don't know if it failed or not.
-                                return (cb(err2));
+                /*
+                 * In some cases, the output from an audit job can be quite
+                 * large.  We don't need to inspect the contents here, just
+                 * report on whether or not the object contained any output.
+                 */
+                var auditOutput = parts[0];
+                MANTA_CLIENT.info(auditOutput, function (infoErr, info) {
+                        if (infoErr) {
+                                cb(VE(infoErr, 'checking audit output "%s"',
+                                    auditOutput));
+                                return;
                         }
-                        if (errorLines !== '') {
-                                //Bad juju.
-                                LOG.fatal({
-                                        job: job,
-                                        outputObject: parts[0]
-                                }, 'Audit job detected abnormalities between ' +
-                                          'mako and moray.');
+
+                        if (typeof (info.size) !== 'number') {
+                                cb(VE('invalid "size" for audit output "%s"',
+                                    auditOutput));
+                                return;
                         }
-                        return (cb(null));
+
+                        if (info.size === 0) {
+                                /*
+                                 * The output object is empty, meaning the
+                                 * audit job did not report any problems.
+                                 */
+                                cb(null);
+                                return;
+                        }
+
+                        /*
+                         * Emit a log message at the FATAL level to trigger
+                         * the appropriate alarm.
+                         */
+                        LOG.fatal({ job: job, outputObject: auditOutput,
+                            outputObjectInfo: info }, 'Audit job detected ' +
+                            'abnormalities between Mako and Moray.');
+                        cb(null);
                 });
         });
 }
-- 
2.21.0

